core/src/test/java/org/apache/iceberg/TestSequenceNumberForV2Table.java

```diff
@@ -0,0 +1,328 @@
+    long snapshotId = table.currentSnapshot().snapshotId();
+
+    table.newFastAppend().appendFile(FILE_B).commit();
+    Assert.assertEquals("Snapshot sequence number should match expected",
+        2, table.currentSnapshot().sequenceNumber());
+
+    table.manageSnapshots().rollbackTo(snapshotId).commit();
+    Assert.assertEquals("Snapshot sequence number should match expected",
+        1, table.currentSnapshot().sequenceNumber());
+
+    Assert.assertEquals("Table last sequence number should be 2",
+        2, table.operations().current().lastSequenceNumber());
+
+    table.newFastAppend().appendFile(FILE_C).commit();
+    Assert.assertEquals("Snapshot sequence number should match expected",
+        3, table.currentSnapshot().sequenceNumber());
+  }
+
+  @Test
+  public void testSingleTransaction() {
+    Transaction txn = table.newTransaction();
+    txn.newAppend().appendFile(FILE_A).commit();
+    txn.commitTransaction();
+    Assert.assertEquals("Snapshot sequence number should match expected",
+        1, table.currentSnapshot().sequenceNumber());
+  }
+
+  @Test
+  public void testConcurrentTransaction() {
+    Transaction txn1 = table.newTransaction();
+    Transaction txn2 = table.newTransaction();
+    Transaction txn3 = table.newTransaction();
+    Transaction txn4 = table.newTransaction();
+
+    txn1.newFastAppend().appendFile(FILE_A).commit();
+    txn3.newOverwrite().addFile(FILE_C).commit();
+    txn4.newDelete().deleteFile(FILE_A).commit();
+    txn2.newAppend().appendFile(FILE_B).commit();
+
+    txn1.commitTransaction();
+    txn2.commitTransaction();
+    txn3.commitTransaction();
+    txn4.commitTransaction();
+
+    Assert.assertEquals("Snapshot sequence number should match expected",
+        4, table.currentSnapshot().sequenceNumber());
+    validateDataFiles(files(FILE_A, FILE_B, FILE_C), seqs(4, 2, 3));
+  }
+
+  @Test
+  public void testMultipleOperationsTransaction() {
+    Transaction txn = table.newTransaction();
+    txn.newAppend().appendFile(FILE_A).commit();
+    Set<DataFile> toAddFiles = new HashSet<>();
+    Set<DataFile> toDeleteFiles = new HashSet<>();
+    toAddFiles.add(FILE_B);
+    toDeleteFiles.add(FILE_A);
+    txn.newRewrite().rewriteFiles(toDeleteFiles, toAddFiles).commit();
+    txn.commitTransaction();
+
+    Assert.assertEquals("Snapshot sequence number should match expected",
+        2, table.currentSnapshot().sequenceNumber());
+    validateDataFiles(files(FILE_A, FILE_B), seqs(2, 2));
+  }
+
+  @Test
+  public void testExpirationInTransaction() {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    long snapshotId = table.currentSnapshot().snapshotId();
+    table.newAppend().appendFile(FILE_B).commit();
+
+    Transaction txn = table.newTransaction();
+    txn.expireSnapshots().expireSnapshotId(snapshotId).commit();
+    txn.commitTransaction();
+
+    Assert.assertEquals("Snapshot sequence number should match expected",
+        2, table.currentSnapshot().sequenceNumber());
```

Comment:
This needs to validate the table's last sequence number.

Reply:
OK.